java/main/src/main/java/com/saucelabs/saucebindings/SauceSession.java

```diff
@@ -231,6 +282,10 @@ public void changeTestName(String name) {
    * @see BaseConfigurations#setTags(List)
    */
   public void addTags(List<String> tags) {
+    if (!isEnabled()) {
```

Comment:
@titusfortner all these conditionals in our methods seem kind of annoying. Is there any way to just turn off the method actions if the extension `!isEnabled()`. A single place to turn on/off the operations, like a fuse box.
However, if we don't have these conditionals then that means that the user has to think about them, right?

Reply:
Right, the idea is to put the ugliness in our code instead of the user needing to put it in theirs. There's definitely a more clever way to delegate the functionality, but I think it would be overkill and also possibly less obvious. We can change the implementation itself later if we need.